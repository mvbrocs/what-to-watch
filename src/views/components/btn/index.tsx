import React, { ReactNode } from 'react';

type BtnProps = {
  children: ReactNode;
  onClick: () => void;
};

const Btn = ({ children, onClick }: BtnProps) => {
  return (
    <button className="btn movie-card__button" type="button" onClick={onClick}>
      {children}
    </button>
  );
};

export default Btn;
