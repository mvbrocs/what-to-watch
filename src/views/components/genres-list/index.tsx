import { MouseEventHandler } from 'react';

export type GenreClickHandler = (genre: string) => MouseEventHandler<HTMLAnchorElement>;
export { GenresList } from './genres-list';
