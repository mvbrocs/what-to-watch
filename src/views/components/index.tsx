export { default as GenresList } from './genres-list';
export { default as Header } from './header';
export { default as Logo } from './logo';
export { default as MovieCard } from './movie-card';
export { default as MovieList } from './movie-list';
export { default as Tabs } from './tabs';
export { default as TabPane } from './tabs/tab-pane';
export { default as UserBlock } from './user-block';
export { default as VideoPlayer } from './video-player';
export { default as VideoPlayerFullscreen } from './video-player-fullscreen';
export { default as Btn } from './btn';
