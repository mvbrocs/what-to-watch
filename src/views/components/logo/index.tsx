import React from 'react';
import { Link } from 'react-router-dom';
import classNames from 'classnames';

import { Path } from 'src/routes';

type LogoProps = {
  light?: boolean;
};

const Logo = ({ light }: LogoProps) => (
  <div className="logo">
    <Link to={Path.HOME} className={classNames('logo__link', { 'logo__link--light': light })}>
      <span className="logo__letter logo__letter--1">W</span>
      <span className="logo__letter logo__letter--2">T</span>
      <span className="logo__letter logo__letter--3">W</span>
    </Link>
  </div>
);

export default Logo;
