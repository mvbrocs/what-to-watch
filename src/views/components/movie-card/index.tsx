export type MovieCardProps = {
  name: string;
  preview_image: string;
  preview_video_link: string;
};
export { MovieCard } from './movie-card';
