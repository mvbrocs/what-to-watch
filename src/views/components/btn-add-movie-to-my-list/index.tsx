import React from 'react';

import { Btn } from 'src/views/components';
import { AddIcon, InListIcon } from 'src/views/icons';
import { IFilm } from 'src/api/films';
import { useBtnAddMovieToMyList } from './use-btn-add-movie-to-my-list';

type BtnAddMovieToMyListProps = {
  movie: IFilm | null;
};

const BtnAddMovieToMyList = ({ movie }: BtnAddMovieToMyListProps) => {
  const { movieInList, clickHandler } = useBtnAddMovieToMyList(movie);
  const btnAddIcon = movieInList ? <InListIcon /> : <AddIcon />;

  return (
    <Btn onClick={clickHandler}>
      {btnAddIcon}
      <span>My list</span>
    </Btn>
  );
};

export default BtnAddMovieToMyList;
