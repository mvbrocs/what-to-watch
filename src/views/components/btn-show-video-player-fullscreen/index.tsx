import React from 'react';

import { Btn } from 'src/views/components/index';
import { PlaySIcon } from 'src/views/icons';
import { useBtnShowVideoPlayerFullscreen } from './use-btn-show-video-player-fullscreen';
import { IFilm } from 'src/api/films';

type BtnShowVideoPlayerFullscreenProps = {
  movie: IFilm | null;
};

const BtnShowVideoPlayerFullscreen = ({ movie }: BtnShowVideoPlayerFullscreenProps) => {
  const { clickHandler } = useBtnShowVideoPlayerFullscreen(movie);

  return (
    <Btn onClick={clickHandler}>
      <PlaySIcon />
      <span>Play</span>
    </Btn>
  );
};

export default BtnShowVideoPlayerFullscreen;
