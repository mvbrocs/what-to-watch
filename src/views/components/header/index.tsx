import React, { ReactNode } from 'react';
import classNames from 'classnames';

import { Logo } from 'src/views/components';

type HeaderProps = {
  className?: string;
  children?: ReactNode;
};

const Header = ({ className, children }: HeaderProps) => (
  <header className={classNames('page-header', className)}>
    <Logo />

    {children}
  </header>
);

export default Header;
