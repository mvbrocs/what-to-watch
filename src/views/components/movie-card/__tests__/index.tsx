import React from 'react';
import { shallow } from 'enzyme';

import { MovieCard } from 'src/views/components';
import { mockFilms } from 'src/mocks';

describe('MovieCard', () => {
  const props = {
    id: mockFilms[0].id,
    name: mockFilms[0].name,
    preview_image: mockFilms[0].preview_image,
    preview_video_link: mockFilms[0].preview_video_link,
  };
  const movieCard = shallow(<MovieCard {...props} />);

  it('renders properly', () => {
    expect(movieCard).toMatchSnapshot();
  });

  it('should render VideoPlayer with play=true prop, on mouseEnter', () => {
    movieCard.simulate('mouseEnter');

    expect(movieCard.find('VideoPlayer').prop('play')).toBeTruthy();
  });

  it('should render VideoPlayer with play=false prop, on mouseLeave', () => {
    movieCard.simulate('mouseLeave');

    expect(movieCard.find('VideoPlayer').prop('play')).toBeFalsy();
  });
});
