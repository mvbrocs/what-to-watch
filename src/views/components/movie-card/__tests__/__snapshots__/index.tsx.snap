// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard renders properly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <VideoPlayer
      delay={1000}
      play={false}
      poster={
        Object {
          "alt": "Orlando",
          "height": 175,
          "src": undefined,
          "width": 280,
        }
      }
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <Link
      className="small-movie-card__link"
      to="/films/1"
    >
      Orlando
    </Link>
  </h3>
</article>
`;
